nixos/modules/services/networking/firewall.nix

```diff
@@ -278,13 +278,35 @@ in
         '';
       };
 
+      defaultInterfaces = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [ "eno1" ];
+        description = ''
+          Restricts the {option}`networking.firewall.allowed*` rules to a
+          specific list of network interfaces.
+
+          If left empty, the rules will apply to all interfaces.
+        '';
+      };
+
       allInterfaces = lib.mkOption {
         internal = true;
         visible = false;
-        default = {
-          default = lib.mapAttrs (name: value: cfg.${name}) commonOptions;
-        }
-        // cfg.interfaces;
+        default =
+          let
+            defaultInterface = lib.optionalAttrs (cfg.defaultInterfaces == [ ]) {
+              default = lib.mapAttrs (option: _: cfg.${option}) commonOptions;
+            };
+
+            defaultInterfaces = lib.genAttrs cfg.defaultInterfaces (
+              interface:
+              lib.mapAttrs (
+                option: _: cfg.${option} ++ cfg.interfaces.${interface}.${option} or [ ]
+              ) commonOptions
+            );
+          in
+          defaultInterface // cfg.interfaces // defaultInterfaces;
```

Comment:
I think this would be much clearer as:

```nix
let
  defaultConfig = lib.mapAttrs (option: _: cfg.${option}) commonOptions;
in
lib.mkMerge (
  [ cfg.interfaces ]
  ++ lib.optional (cfg.defaultInterfaces == [ ]) { default = defaultConfig; }
  ++ lib.optional (cfg.defaultInterfaces != [ ]) (lib.genAttrs cfg.defaultInterfaces (_: defaultConfig))
);
```

Reply:
Unfortunately I don't think it's possible to use `mkMerge` in default option values. Guess I could configure `allInterfaces` in the `config` block like this:

```nix
networking.firewall.allInterfaces =
  let
    defaultConfig = lib.mapAttrs (option: _: cfg.${option}) commonOptions;
  in
  lib.mkMerge (
    lib.optional (cfg.defaultInterfaces == [ ]) { default = defaultConfig; }
    ++ [ cfg.interfaces ]
    ++ [ (lib.genAttrs cfg.defaultInterfaces (_: defaultConfig)) ]
  );
```

But I'm not sure this is preferable as it splits things up and the `allInterfaces` value would no longer have `mkDefault` ranking (not sure this matters)

Reply:
You can do `lib.mkOptionDefault (lib.mkMerge ...)`

Reply:
- Knowing that the suggestion I wrote doesn't work, I totally rescind my comment. This is fine as-is
- We don't need to worry about the rank at all; This is an option marked `internal = true` and `visible = false`, and a grep thru nixpkgs shows it's only set here. This option should probably have `readOnly = true`.
- Separating the option and config (within the same file but a bunch of lines apart) is fine; that's just how nix modules are
- If you really wanted to keep them close together, you can add an inline module that only contains one option:

```nix
imports = [ {
  options.networking.firewall.allInterfaces = lib.mkOption { /* ... */ };
  config.networking.firewall.allInterfaces = lib.mkMerge [ /* ... */ ];
} ];
```